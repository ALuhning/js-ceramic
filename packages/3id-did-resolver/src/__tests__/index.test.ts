import ThreeIdResolver from '../index'
import { Resolver } from 'did-resolver'

const ceramicMock = {
  loadDocument: async (): Promise<any> => ({
    content: {
      publicKeys: {
        signing: 'zQ3shwsCgFanBax6UiaLu1oGvM7vhuqoW88VBUiUTCeHbTeTV',
        encryption: 'z6LSfQabSbJzX8WAm1qdQcHCHTzVv8a2u6F7kmzdodfvUCo9'
      }
    }
  })
}

const ceramicMockOld = { // to be removed
  loadDocument: async (): Promise<any> => ({
    content: {
      publicKeys: {
        signing: 'fake signing key',
        encryption: 'fake encryption key'
      }
    }
  })
}

describe('3ID DID Resolver', () => {

  it('getResolver works correctly', async () => {
    const threeIdResolver = ThreeIdResolver.getResolver(ceramicMock)
    expect(Object.keys(threeIdResolver)).toEqual(['3'])
  })

  it('resolver works correctly (old format)', async () => {
    const threeIdResolver = ThreeIdResolver.getResolver(ceramicMockOld)
    const resolver = new Resolver(threeIdResolver)
    const fake3ID = 'did:3:bafyiuh3f97hqef97h'
    expect(await resolver.resolve(fake3ID)).toMatchSnapshot()
  })

  it('resolves 3id document correctly', async () => {
    const threeIdResolver = ThreeIdResolver.getResolver(ceramicMock)
    const resolver = new Resolver(threeIdResolver)
    const fake3ID = 'did:3:bafyiuh3f97hqef97h'
    expect(await resolver.resolve(fake3ID)).toMatchSnapshot()
  })
})
