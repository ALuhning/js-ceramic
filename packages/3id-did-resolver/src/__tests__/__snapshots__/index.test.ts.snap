// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`3ID DID Resolver Legacy (v0) resolves 3id v0 1`] = `
Object {
  "@context": "https://w3id.org/did/v1",
  "authentication": Array [
    Object {
      "publicKey": "did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi#signingKey",
      "type": "Secp256k1SignatureAuthentication2018",
    },
  ],
  "id": "did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi",
  "publicKey": Array [
    Object {
      "id": "did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi#signingKey",
      "publicKeyHex": "0452fbcde75f7ddd7cff18767e2b5536211f500ad474c15da8e74577a573e7a346f2192ef49a5aa0552c41f181a7950af3afdb93cafcbff18156943e3ba312e5b2",
      "type": "Secp256k1VerificationKey2018",
    },
    Object {
      "id": "did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi#encryptionKey",
      "publicKeyBase64": "DFxR24MNHVxEDAdL2f6pPEwNDJ2p0Ldyjoo7y/ItLDc=",
      "type": "Curve25519EncryptionPublicKey",
    },
    Object {
      "ethereumAddress": "0x3f0bb6247d647a30f310025662b29e6fa382b61d",
      "id": "did:3:bafyreiffkeeq4wq2htejqla2is5ognligi4lvjhwrpqpl2kazjdoecmugi#managementKey",
      "type": "Secp256k1VerificationKey2018",
    },
  ],
}
`;

exports[`3ID DID Resolver Legacy (v0) resolves 3id v0 which includs ceramic updates 1`] = `
Object {
  "@context": "https://w3id.org/did/v1",
  "authentication": Array [
    Object {
      "publicKey": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#signing",
      "type": "Secp256k1SignatureAuthentication2018",
    },
  ],
  "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
  "keyAgreement": Array [
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#encryption",
      "publicKeyBase58": "4jQRvHW8RfnRfdTrsxmExsn24z2vCV4xsoGxKB2Pkq2P",
      "type": "X25519KeyAgreementKey2019",
    },
  ],
  "publicKey": Array [
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#signing",
      "publicKeyBase58": "2Bv3fQSJiGJsbUUfUdiPsXXZGG5cXqaSeYnieuJcUFAbm",
      "type": "Secp256k1VerificationKey2018",
    },
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#encryption",
      "publicKeyBase58": "4jQRvHW8RfnRfdTrsxmExsn24z2vCV4xsoGxKB2Pkq2P",
      "type": "Curve25519EncryptionPublicKey",
    },
  ],
}
`;

exports[`3ID DID Resolver adds IDX root as service 1`] = `
Object {
  "@context": "https://w3id.org/did/v1",
  "authentication": Array [
    Object {
      "publicKey": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#signing",
      "type": "Secp256k1SignatureAuthentication2018",
    },
  ],
  "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
  "keyAgreement": Array [
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#encryption",
      "publicKeyBase58": "4jQRvHW8RfnRfdTrsxmExsn24z2vCV4xsoGxKB2Pkq2P",
      "type": "X25519KeyAgreementKey2019",
    },
  ],
  "publicKey": Array [
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#signing",
      "publicKeyBase58": "2Bv3fQSJiGJsbUUfUdiPsXXZGG5cXqaSeYnieuJcUFAbm",
      "type": "Secp256k1VerificationKey2018",
    },
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#encryption",
      "publicKeyBase58": "4jQRvHW8RfnRfdTrsxmExsn24z2vCV4xsoGxKB2Pkq2P",
      "type": "Curve25519EncryptionPublicKey",
    },
  ],
  "service": Array [
    Object {
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#idx",
      "serviceEndpoint": "ceramic://rootId",
      "type": "IdentityIndexRoot",
    },
  ],
}
`;

exports[`3ID DID Resolver resolves 3id document correctly 1`] = `
Object {
  "@context": "https://w3id.org/did/v1",
  "authentication": Array [
    Object {
      "publicKey": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#signing",
      "type": "Secp256k1SignatureAuthentication2018",
    },
  ],
  "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
  "keyAgreement": Array [
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#encryption",
      "publicKeyBase58": "4jQRvHW8RfnRfdTrsxmExsn24z2vCV4xsoGxKB2Pkq2P",
      "type": "X25519KeyAgreementKey2019",
    },
  ],
  "publicKey": Array [
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#signing",
      "publicKeyBase58": "2Bv3fQSJiGJsbUUfUdiPsXXZGG5cXqaSeYnieuJcUFAbm",
      "type": "Secp256k1VerificationKey2018",
    },
    Object {
      "controller": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki",
      "id": "did:3:k2t6wyfsu4pg0t2n4j8ms3s33xsgqjhtto04mvq8w5a2v5xo48idyz38l7ydki#encryption",
      "publicKeyBase58": "4jQRvHW8RfnRfdTrsxmExsn24z2vCV4xsoGxKB2Pkq2P",
      "type": "Curve25519EncryptionPublicKey",
    },
  ],
}
`;
